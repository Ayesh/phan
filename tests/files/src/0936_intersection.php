<?php

interface I936 {
    public function returnsString(): string;
}
function test_intersection_936(ArrayAccess $t) {
    if ($t instanceof I936) {
        '@phan-debug-var $t';
        echo strlen($t->count()); // not defined, that requires Countable.
        $t->offsetSet('key', 'value');
        echo intdiv($t->returnsString(), 2);
    }
    echo "After merging union types\n";
    '@phan-debug-var $t';
}

trait T936 {
    public function myMethod() {
        if ($this instanceof C936) {
            '@phan-debug-var $this';
            $this->otherTraitMethod(123);
            var_dump($this->count());
        }
    }
    public function otherTraitMethod(int $arg) {
        echo $arg, "\n";
    }
}

class C936 extends ArrayObject {
    use T936;
}
