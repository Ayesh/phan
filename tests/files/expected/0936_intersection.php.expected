%s:7 PhanDebugAnnotation @phan-debug-var requested for variable $t - it has union type \ArrayAccess&\I936(real=\ArrayAccess&\I936)
%s:9 PhanUndeclaredMethod Call to undeclared method \ArrayAccess::count
%s:11 PhanTypeMismatchArgumentInternal Argument 1 ($num1) is $t->returnsString() of type string but \intdiv() takes int
%s:13 PhanDebugAnnotation @phan-debug-var requested for variable $t - it has union type \ArrayAccess|\ArrayAccess&\I936(real=\ArrayAccess|\ArrayAccess&\I936)
%s:19 PhanDebugAnnotation @phan-debug-var requested for variable $this - it has union type static&\C936(real=static&\C936)
